lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -550,200 +425,246 @@ export class UserAgentApplication {
+      if (Utils.isSSOParam(request) || account) {
         serverAuthenticationRequest = this.populateQueryParams(account, request, serverAuthenticationRequest);
+      }
+      //if user didn't pass login_hint/sid and adal's idtoken is present, extract the login_hint from the adalIdToken
+      else if (!account && !Utils.isEmpty(adalIdToken)) {
+        // if adalIdToken exists, extract the SSO info from the same
+        const adalIdTokenObject = Utils.extractIdToken(adalIdToken);
+        this.logger.verbose("ADAL's idToken exists. Extracting login information from ADAL's idToken ");
+        serverAuthenticationRequest = this.populateQueryParams(account, null, serverAuthenticationRequest, adalIdTokenObject);
+      }
+      let userContainedClaims = request.claimsRequest || serverAuthenticationRequest.claimsValue;
 
-        this.updateCacheEntries(serverAuthenticationRequest, account);
-
-        // Construct the urlNavigate
-        let urlNavigate = serverAuthenticationRequest.createNavigateUrl(request.scopes) + Constants.response_mode_fragment;
-
-        window.renewStates.push(serverAuthenticationRequest.state);
-        window.requestType = Constants.renewToken;
-        this.registerCallback(serverAuthenticationRequest.state, scope, resolve, reject);
+      let authErr: AuthError;
+      let cacheResultResponse;
 
-        // open popup window to urlNavigate
-        if (popUpWindow) {
-          popUpWindow.location.href = urlNavigate;
+      if (!userContainedClaims) {
+        try {
+          cacheResultResponse = this.getCachedToken(serverAuthenticationRequest, account);
+        } catch (e) {
+          authErr = e;
         }
+      }
 
-      }, () => {
-        // Endpoint resolution failure error
-        this.logger.info(ClientAuthErrorMessage.endpointResolutionError.code + ":" + ClientAuthErrorMessage.endpointResolutionError.desc);
-        this.cacheStorage.setItem(Constants.msalError, ClientAuthErrorMessage.endpointResolutionError.code);
-        this.cacheStorage.setItem(Constants.msalErrorDescription, ClientAuthErrorMessage.endpointResolutionError.desc);
-
-        // reject that is passed in - REDO this in the subsequent refactor, passing reject is confusing
-        if (reject) {
-          reject(ClientAuthError.createEndpointResolutionError());
+      // resolve/reject based on cacheResult
+      if (cacheResultResponse) {
+        this.logger.info("Token is already in cache for scope:" + scope);
+        resolve(cacheResultResponse);
+        return null;
+      }
+      else if (authErr) {
+        this.logger.infoPii(authErr.errorCode + ":" + authErr.errorMessage);
+        reject(authErr);
+        return null;
+      }
+      // else proceed with login
+      else {
+        if (userContainedClaims) {
+          this.logger.verbose("Skipped cache lookup since claims were given.");
+        } else {
+          this.logger.verbose("Token is not in cache for scope:" + scope);
```

Comment:
The value of the log message could be created as a ternary.

Reply:
done